cli/command/service/update.go

```diff
@@ -164,9 +164,16 @@ func runUpdate(dockerCli *command.DockerCli, flags *pflag.FlagSet, options *serv
 	}
 
 	if flags.Changed("image") {
-		if err := resolveServiceImageDigest(dockerCli, spec); err != nil {
+		if err := resolveServiceImageDigestContentTrust(dockerCli, spec); err != nil {
 			return err
 		}
+		noResolveImage, err := flags.GetBool(flagNoResolveImage)
+		if err != nil {
+			return err
+		}
+		if !noResolveImage {
```

Comment:
The `options` struct is the preferred way of doing it. You shouldn't need to use `flags.GetBool()` in most cases.